File: typescript/Structural/12-Proxy/Model/JobCard.ts
const shortId = () => Math.random().toString(36).slice(2, 10);

export class JobCard {
  readonly jobId: string = shortId();
  status: string = "Active";

  constructor(
    readonly title: string,
    readonly category: string,
  ) {}

  toString(): string {
    return `[${this.category}] #${this.jobId} — ${this.title} (${this.status})`;
  }
}

File: csharp/Structural/12-Proxy/Model/JobCard.cs
public class JobCard
{
    public string JobId { get; init; } = Guid.NewGuid().ToString()[..8];
    public string Title { get; init; } = string.Empty;
    public string Category { get; init; } = string.Empty;
    public string Status { get; set; } = "Active";

    public override string ToString() =>
        $"[{Category}] #{JobId} — {Title} ({Status})";
}
